lib/tasks/repair_nfn.rake

```diff
@@ -0,0 +1,43 @@
+namespace :data_repair do
+  task :nfn do
+    query = %q(
+      select user_id, reducer_key, subgroup
+      from user_reductions
+      where reducible_type='Project' and reducible_id=1558
+      group by reducible_type, reducible_id, reducer_key, user_id, subgroup
+      having count(*) > 1
+    )
+
+    affected_items = ActiveRecord::Base::Connection.execute(query)
+
+    affected_items.each do |params|
+      user_id,  reducer_key, subgroup = params
+
+      ActiveRecord::Base.transaction do
+        reductions = UserReductions.where(
+          reducible_type: 'Project',
+          reducible_id: 1558,
+          user_id: user_id,
+          reducer_key: reducer_key,
+          subgroup: subgroup
+        ).lock(true)
+
+        data = {
+          extracts: reductions.inject(0){ |sum, r| sum + r.data['extracts'] },
+          classifications: reductions.inject(0){ |sum, r| sum + r.data['classifications'] },
+        }
+
+        reductions.delete_all
```

Comment:
We're not uniquing yet so you could create the new reduction before deleting the old ones, in case something fails.

Reply:
Since it's all done in a transaction, if something goes wrong the deletes will be rolled back as well :)

Reply:
i ended up needing to do this anyways as it turned out!